src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentPreprocessorDirectiveTest.cs

```diff
@@ -0,0 +1,727 @@
+        var generated = CompileToCSharp("""
+            @{
+            #if false
+                <p>Some text</p>
+            #endif
+            }
+            """);
+
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated);
+    }
+
+    [IntegrationTestFact]
+    public void PassParseOptionsThrough_01()
+    {
+        var parseOptions = CSharpParseOptions.WithPreprocessorSymbols("SomeSymbol");
+
+        var generated = CompileToCSharp("""
+            @{
+            #if SomeSymbol
+                <p>Some text</p>
+            #endif
+            }
+            """,
+            csharpParseOptions: parseOptions);
+
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated);
+    }
+
+    [IntegrationTestFact]
+    public void PassParseOptionsThrough_02()
+    {
+        var parseOptions = CSharpParseOptions.WithPreprocessorSymbols("SomeSymbol");
+
+        var generated = CompileToCSharp("""
+            @{
+            #if !SomeSymbol
+                <p>Some text</p>
+            #endif
+            }
+            """,
+            csharpParseOptions: parseOptions);
+
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated);
+    }
+
+    [IntegrationTestFact]
+    public void DefineAndUndef()
+    {
+        var generated = CompileToCSharp("""
+            @{
+            #define SomeSymbol
+            #undef SomeSymbol
+            }
+            """);
+
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+        CompileToAssembly(generated,
+            // x:\dir\subdir\Test\TestComponent.cshtml(2,2): error CS1032: Cannot define/undefine preprocessor symbols after first token in file
+            // #define SomeSymbol
+            Diagnostic(ErrorCode.ERR_PPDefFollowsToken, "define").WithLocation(2, 2),
+            // x:\dir\subdir\Test\TestComponent.cshtml(3,2): error CS1032: Cannot define/undefine preprocessor symbols after first token in file
+            // #undef SomeSymbol
+            Diagnostic(ErrorCode.ERR_PPDefFollowsToken, "undef").WithLocation(3, 2)
+        );
+    }
+
+    [IntegrationTestFact]
+    public void AfterTag()
+    {
+        var generated = CompileToCSharp("""
+            @{
+                <div>
+            #if true
+                <div>
+            }
+            @{
+            #endif
+            }
+            """);
+
+        AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+        if (!DesignTime)
+        {
+            // https://github.com/dotnet/razor/issues/10981
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
```

Comment:
We have parameter `verifyLinePragmas`, consider using just that instead of disabling the whole C# document baseline verification.